app/models/MessageType.scala

```diff
@@ -62,6 +62,83 @@ object MessageType extends Enumerable.Implicits {
         values.find(_.rootNode == nodeSeq.head.label)
     }
 
+  implicit val ordering: Ordering[MessageType] = (x: MessageType, y: MessageType) => {
```

Comment:
I'm kind of nervous about this for a couple of reasons: 

* This is brittle. I don't see any problems with what you've written in that I can see you'll get a consistent ordering every time you use this, but in the very unlikely event that we ever need to add another status (if this gets reused for future phases, I guess?) or we want to rejig the order for some reason, then we have to go through all of this - and it's not guaranteed that anyone will remember to do so or that the written tests will pick it up - there are always edge cases!
* I _think_ this compiles to a set of if/else statements in the bytecode (unfortunately, it can't compile to lookupswitch and tableswitch instructions as they only operates on ints) and if you're ordering any sort of list where, say, `XMLSubmissionNegativeAcknowledgement` is being compared against something, a _lot_ of unnecessary comparisons could be made. I'm not sure this is going to get called all that much, but I'd rather avoid any potential bottlenecks where possible. 

Would it be better if we give each of the `MessageType`s here an integer ordering parameter and just compare on that? The JVM is used to comparing numbers and I suspect that it'll be able to optimise these comparisons far better if we can do that.

Reply:
That makes sense. I didn't actually write this, I moved it from the frontend, but I'll take a closer look at it

Reply:
We've re-written this. The id word fine apart from one edge case where one status' position in the list is variable. But I've dealt with that in a case statement

Reply:
Sorry to bikeshed this even more but I think the entire declaration can now be replaced by
```scala
implicit val ordering: Ordering[MessageType] = Ordering.by(_.id)
```

Reply:
It can't. The problem is XMLSubmissionNegativeAcknowledgement can be returned after both DepartureDeclaration and DeclarationCancellationRequest, which aren't next to each other in the priority order

So if we get either of those status with an XMLSubmission... then we return XMLSubmission...
But if we get XMLSubmission... with any other status, then we return that.

So I placed XMLSubmission... after DepartureDeclaration and had to add a match statement for the one edge case of DeclarationCancellationRequest and XMLSubmission...

Reply:
Oh I see, I couldn't see those special cases in the original diff